torch/csrc/autograd/engine.cpp

```diff
@@ -59,6 +59,7 @@ auto Engine::backward(const variable_list& variables,
                       tensor_list& grad_variables,
                       bool retain_variables) -> void {
   function_queue creators;
+  std::unordered_map<std::shared_ptr<Function>, std::unique_ptr<GradBuffer>> creator_grad;
```

Comment:
No you can't. I later need to insert shared pointers to `ready`, so I need to keep them in the map too.